### PR TITLE
wolfi-images: add mi2 base image

### PR DESCRIPTION
**Pull request by michaellzc**
eventually replace https://github.com/sourcegraph/controller/blob/main/cmd/mi2/Dockerfile

### Test plan

CI

### CODE REVIEW

File: wolfi-images/cloud-mi2.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    # - terraform
+    # - helm
+    # - kustomize
+    # - nodejs-18
+    # TODO: pinned versions work locally
+    # but it's broken in CI?
+    - terraform=1.3.9
+    - helm=3.11.3
+    - kustomize=4.5.7
+    - nodejs-18=18.17.1
```
Comment (michaellzc) [inline wolfi-images/cloud-mi2.yaml:18]: broken build https://buildkite.com/sourcegraph/sourcegraph/builds/238957#018a0241-1931-48ee-a2fb-40d40fd2084a

File: wolfi-images/cloud-mi2.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+include: ./sourcegraph-base.yaml
+
+contents:
+  packages:
+    # Included by existing SG base image
+    - tini
+    - mailcap
+    - terraform=1.3.9-r2
```
Comment (willdollman) [inline wolfi-images/cloud-mi2.yaml:8]:
```suggestion

    ## mi2 packages
    - terraform=1.3.9-r2
```

Added a comment above mi2-specific packages, just for consistency with other base images

**Review by willdollman — APPROVED** [review]
lgtm. Out of curiosity, are you planning on using this as-is or adding a `docker-images/mi2/BUILD.bazel` to build a full image with `mi2` installed?

Edit: Ah, just spotted https://github.com/sourcegraph/controller/blob/main/cmd/mi2/Dockerfile :)

**Comment by michaellzc** [comment]
> lgtm. Out of curiosity, are you planning on using this as-is or adding a `docker-images/mi2/BUILD.bazel` to build a full image with `mi2` installed?
> 
> Edit: Ah, just spotted https://github.com/sourcegraph/controller/blob/main/cmd/mi2/Dockerfile :)

it will be bazel eventually https://github.com/sourcegraph/controller/pull/962

we run into [some blockers](https://sourcegraph.slack.com/archives/C05FDP58LRF/p1692413675195059) rolling out bazel in this repo, so it will take some time

**Pull request merged**